### DISCUSSION TITLE
Pressing the red X closes ALL tabs in ALL Workspaces (not expected by the user)

### DISCUSSION THREAD

**Discussion by trendespresso** [zen-browser/desktop#1331] [Ideas]
Hiya!

I'd like to start by saying I've really been enjoying Zen Browser. It's definitely my favourite overall compared to Chrome, Brave, Firefox, Arc, and Sigma. However, the primary reason why I can't make it my daily driver is because of the poor tab permanence.

At any given moment, Zen can decide to zoot all my tabs. I've had all my tabs lost when restarting my computer, letting my computer go to sleep for more than a few hours, or pressing the red circled X in the top left.

Example of two bugs:
[https://youtu.be/jnHoNV3w_Qk](https://youtu.be/jnHoNV3w_Qk)

1. When opening a tab – or especially if opening multiple tabs quickly – switching Workspaces while any tab is loading results in the tab's parent Workspace becoming the newly switched Workspace rather than the one selected whilst the tab was originally opened.
2. Pressing the red circled X in the top left closes all tabs!

The latter is especially damning because the user would logically expect that pressing the red circled X in the top left of the window would either A) Close all the tabs on the current Workspace but leave all other Workspaces alone, or B) Close the window and save all opened tabs so long as Preferences -> Startup -> Open previous windows and tabs is checked.

At least the red circled X behaviour needs to be changed.

Thank you for your time and effort in working on Zen!
Cheers

**Comment by universish** [comment]
This can be added as a toolbar button. A new toolbar feature could be developed. Right click > Customize toolbar, drag the symbol, drop it on the toolbar.
There is already a button. Is this what you are talking about:
A symbol called “Forget part of my navigation history” (there is a clock symbol, around the clock there is an arrow pointing counterclockwise, and on the clock symbol there is a diagonal line, this is the shape of the symbol).

**Reply by trendespresso** [reply]
I'm not sure I follow? Could you provide a screenshot?

I think my video shows the first bug quite well. For the second issue, simply open a bunch of tabs and press the red circled X in the top left. I'm using macOS. All tabs close and are not saved in any workspace despite that being the intuitive behaviour.

**Reply by universish** [reply]
I didn't watch your video. I just watched it. I see what you're talking about.

**Comment by tech-leon** [comment]
I left Zen because of this issue. It's too painful when all my tags are gone, even the setting "Open previous windows and tabs." is on.

**Comment by mausv** [comment]
Seeing this issue as well on 1.7.6b.

**Reply by hzq23x** [reply]
still persisting on 1.11.5b. love the time put in and work the team has done and enjoying the browser but this one issue is bothering me. hopefully they fix it.
